quinn-proto/src/packet.rs

```diff
@@ -92,20 +82,79 @@ impl PartialDecode {
         self,
         header_crypto: Option<&HeaderCrypto>,
     ) -> Result<(Packet, Option<BytesMut>), PacketDecodeError> {
+        use self::PlainHeader::*;
         let Self {
-            invariant_header,
+            plain_header,
             mut buf,
         } = self;
-        let (payload_len, header, allow_coalesced) = match invariant_header {
-            InvariantHeader::Short { dst_cid, .. } => {
+        let (payload_len, header, allow_coalesced) = match plain_header {
+            Initial {
+                dst_cid,
+                src_cid,
+                token,
+                len,
+            } => {
+                let number = Self::decrypt_header(&mut buf, header_crypto.unwrap())?;
+                if len < number.len() as u64 {
+                    return Err(PacketDecodeError::InvalidHeader(
+                        "packet number longer than packet",
+                    ));
+                }
```

Comment:
Let's get rid of this check while we're in here, since it's subsumed by decode_header.